src/main.c

```diff
@@ -1,6 +1,182 @@
+/**
+ * \brief Main file for Sensor Fusion Algorithm software
+ * \author jaspalsingh@cmail.carleton.ca
+ * \version 1.0
+ * \date 2019-12-05
+ */
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <libgen.h>
+#include <string.h>
+#include <limits.h>
+#include <time.h>
+#include "log_handling.h"
+#include "sensor_fusion.h"
 
-int main(void) {
-  // code goes here
-  return 0;
+#define BUF_LEN 255
+extern int errno ;
+
+void get_curr_timestamp(char *time_now) {
+    //Refer: http://zetcode.com/articles/cdatetime/
+    time_t rawtime = time(NULL);
+    if (rawtime == -1) {
+        puts("The time() function failed");
+        exit(EXIT_FAILURE);
+    }
+    struct tm *ptm = localtime(&rawtime);
+    if (ptm == NULL) {
+        puts("The localtime() function failed");
+        exit(EXIT_FAILURE);
+    }
+    strftime(time_now, BUF_LEN, "%Y%m%d-%H%M%S", ptm);
 }
+
+int main(int argc, char *argv[]) {
+    char *input_csv = NULL;
+    char *output_csv = NULL;
+    char log_path[BUF_LEN];
+    char *proj_abs_path;
+    FILE *fptr_log;
+    FILE *fptr_csv_out;
+    int errnum = 0;
```

Comment:
As per the discussion in the forum regarding the coding conventions while naming variables, it is recommended to use snake style .Therefore, the variable 'errnum' could be replaced with 'err_num'

Reply:
Replace errnum with err_num in the latest commit